pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java

```diff
@@ -259,6 +271,24 @@ protected void map(LongWritable key, Text value, Context context)
         sequenceId);
   }
 
+  /**
+   * Generate an output directory path for bootstrap mode.
+   * This method will compute the relative path based on `inputFile` and `baseInputDir`,
+   * then apply only the directory part of relative path to `outputDir`.
+   * E.g.
+   *    baseInputDir = "/path/to/input"
+   *    inputFile = "/path/to/input/a/b/c/d.avro"
+   *    outputDir = "/path/to/output"
+   *    getBootstrapOutputPath(baseInputDir, inputFile, outputDir) = /path/to/output/a/b/c
+   */
+  protected static Path getBootstrapOutputPath(URI baseInputDir, URI inputFile, Path outputDir) {
+    URI relativePath = baseInputDir.relativize(inputFile);
+    if (relativePath.getPath().length() > 0) {
+      return new Path(outputDir, relativePath.getPath()).getParent();
+    }
+    return null;
```

Comment:
In what scenario can this return null? Can it happen in normal case? Should throw exception instead IMO

Reply:
good point. Change this to throw RTE